redash/models/__init__.py

```diff
@@ -232,17 +232,35 @@ class DataSourceGroup(db.Model):
     __tablename__ = "data_source_groups"
 
 
+class DBPersistence(object):
+    @property
+    def data(self):
+        if self._data is None:
```

Comment:
Shouldn't this also check if `self._deserialized_data` has already been set and return it if yes (and also unset it in the property setter below)?

Reply:
In what case `_data` will be `None` and `_deserialized_data` has some value?

Reply:
This is basically a race condition when `_data` is set to `None` via the data property setter below when it was previously not `None`. If `data` should just be used as a read-only property I would suggest to remove the property setter.

Reply:
I don't know. It feels like this complicates the code for no good reason, as this class has a very defined use :thinking:

Reply:
Read your comment again and realized that actually making `_deserialized_data` in sync with the setter isn't that complex. Implemented in 16fbc5e2940fc6b98d2bfdc53653479347ba403d + added tests.